Make `useDisplayTaxRate` check also store's country

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Move `useStoreCountry` hook to the shared folder,
- Make `useDisplayTaxRate` check also store's country, but take audience countries as an argument.
- Stop exporting `shouldDisplayTaxRate` as nothing uses it any longer, YAGNI.

Fixes: https://github.com/woocommerce/google-listings-and-ads/issues/313.


### Screenshots:

![Check store country](https://user-images.githubusercontent.com/17435/114939150-54a30f00-9e40-11eb-9e54-70406fe8f735.gif)

This fix touched the code responsible for showing/hiding unsupported country banner:
![Supported store country](https://user-images.githubusercontent.com/17435/114940070-92546780-9e41-11eb-988e-72c5828ba328.png)

![Unsupported store country](https://user-images.githubusercontent.com/17435/114940079-954f5800-9e41-11eb-8ba8-23ac5b3203d1.png)



### Detailed test instructions:

#### US in the audience or as a store's country in Setup MC 
1. Set store's country to non-US in [Woo settings](https://gla1.test/wp-admin/admin.php?page=wc-settings)
1. (Re-)Start [MC Setup flow](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc)
2. On "Choose your audience" select country/-ies other than the US
3. Continue to the next step
4. Make sure no Tax Rate section is **not** shown
5. Go back to "Choose your audience", select "all" or pick the US from the list.
6. Continue to the next step
7. Make sure no Tax Rate section is shown
8. Change store's country to the US in [Woo settings](https://gla1.test/wp-admin/admin.php?page=wc-settings)
9. open [MC Setup flow](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc)
2. On "Choose your audience" select country/-ies other than the US
3. Continue to the next step
4. Make sure no Tax Rate section is shown
5. Go back to "Choose your audience" select "all" or pick the US from the list.
6. Continue to the next step
7. Make sure no Tax Rate section is shown

#### US in the audience or as a store's country in edit free listings
Do the above by with [Edit free listings page](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fedit-free-campaign&programId=0&pageStep=1) 

#### Regression of unsupported notice
1. With supported store country open [Start page](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart)
2. Make sure no "not supported" banner is shown.
3. Change your country to an unsupported one, or put `return false;` at https://github.com/woocommerce/google-listings-and-ads/blob/fix/313-store-country-tax/src/MerchantCenter/MerchantCenterService.php#L83:L83
2. Make sure the "not supported" banner is shown.



### Changelog Note:

> Check also store location before showing tax settings.

### Additional notes

-I decided to use a data hook [in the middle of the form](https://github.com/woocommerce/google-listings-and-ads/pull/491/commits/05a8b378be52fdf4c13c211d8432838a1aca3702#diff-5f87cc11620c6da6cadd69a3a7a57f4e7337871d3d709f4bcd427f1b8ff1e0afR39), as we are fetching the WC store's configuration, which is unlikely to be variable in our plugin.

tomalec committed:
Move `useStoreCountry` hook to shared folder,

from `UnsupportedCountry` component.
Make it return both country name and code.
This hook will be later used to check whether the Tax Rate section should be displayed.

tomalec committed:
Make `useDisplayTaxRate` check also store's country,

but take a audience countries as an argument.

Stop exporting `shouldDisplayTaxRate` as nothing uses it any longer, YAGNI.

Fixes: https://github.com/woocommerce/google-listings-and-ads/issues/313.

tomalec committed:
Fix `useDisplayTaxRate` condition to cater for `null` state.

For example when audience countries are still resolving.
Make it return `null` if we cannot tell yet due to some data being unresolved and the other not containing `US`.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/491#discussion_r614528931.

tomalec committed:
Show preloader when TaxRate section is not resolved yet.

tomalec committed:
Add tests for `useDisplayTaxRate`.

eason9487 reviewed (approved):
Tested it locally, works well. Left a comment for polishing and good to go.

tomalec committed:
Remove comented out `console.log`.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/491#discussion_r618041851.
